### DISCUSSION TITLE
zlib as wasm module for use in web applications

### DISCUSSION THREAD

**Discussion by elrido** [madler/zlib#938] [Show and tell]
Not really something I made, but I now [maintain a fork of someone else's git repo](https://github.com/PrivateBin/zlib-wasm-without-emscripten-sample) that provides tooling to build zlib into a web assembly module using nodejs/npm. [We used it in a project](https://github.com/PrivateBin/PrivateBin/blob/master/js/zlib-1.3.1.js) to replace a JavaScript deflate implementation that had several bugs over the years. We considered switching to another, better supported JavaScript implementation (pako), but why risk further regressions when you can use the reference implementation.

Has anyone else been using zlib in web applications on the client side?

Should there be any interest and if we could figure out a way to get zlib transpiled into wasm using [cmake and GitHub actions](https://github.com/madler/zlib/blob/develop/.github/workflows/cmake.yml), would it be an option to produce and distribute such a target from this project?

**Comment by nmoinvaz** [comment]
Zlib-ng supports enscripten.
